src/ray/intersect_x86_64.rs

```diff
@@ -0,0 +1,283 @@
+use std::arch::x86_64::*;
+
+use nalgebra::SVector;
+
+use crate::{aabb::Aabb, utils::fast_max};
+
+use super::{intersect_default::RayIntersection, Ray};
+
+trait ToRegisterType {
+    type Register;
+
+    fn to_register(&self) -> Self::Register;
+}
+
+impl ToRegisterType for SVector<f32, 2> {
+    type Register = __m128;
+
+    #[inline(always)]
+    fn to_register(&self) -> Self::Register {
+        unsafe { _mm_set_ps(self.y, self.y, self.y, self.x) }
+    }
+}
+
+impl ToRegisterType for SVector<f32, 3> {
+    type Register = __m128;
+
+    #[inline(always)]
+    fn to_register(&self) -> Self::Register {
+        unsafe { _mm_set_ps(self.z, self.z, self.y, self.x) }
+    }
+}
+
+impl ToRegisterType for SVector<f32, 4> {
+    type Register = __m128;
+
+    #[inline(always)]
+    fn to_register(&self) -> Self::Register {
+        unsafe { _mm_set_ps(self.w, self.z, self.y, self.x) }
+    }
+}
+
+#[inline(always)]
+fn max_elem_m128(mm: __m128) -> f32 {
```

Comment:
Could you write a short blip of doc comment for each of these functions? I think it would make the file easier to scan.